### DISCUSSION TITLE
Did firmware upgrade on CPC200-Autokit now it crashes.

### DISCUSSION THREAD

**Discussion by dogzipp** [tesla-android/issue-tracker#66] [Q&A]
Autokit prompted there was a firmware upgrade, I figured why not, and did it. Afterwards, it will not start up anymore. Seems autokit keeps crashing and crashing on the Android rPI... Tried 3 times a fresh 'etch' of the Android rPI with same results.

**Comment by mikegapinski** [comment]
It seems like there is some kind of issue with the dongle, contact Carlinkit, they’ll ship you a new one. 

There is also a way to manually restore the firmware, they’ll let you know how to do it.

**Comment by dogzipp** [comment]
I've asked them if there's a way to downgrade the firmware, I'll wait for a reply from them.

**Reply by mikegapinski** [reply]
I didn’t mean that you’d need to downgrade. Maybe the install got corrupted when you were updating. I am running the latest firmware without any issues
